pkg/cmd/dockerregistry/dockerregistry.go

```diff
@@ -242,3 +248,31 @@ func panicHandler(handler http.Handler) http.Handler {
 		handler.ServeHTTP(w, r)
 	})
 }
+
+func setDefaultMiddleware(config *configuration.Configuration) (changed bool, err error) {
+	// Default to openshift middleware for relevant types
+	// This allows custom configs based on old default configs to continue to work
+	if config.Middleware == nil {
+		config.Middleware = map[string][]configuration.Middleware{}
+	}
+	for _, middlewareType := range []string{"registry", "repository", "storage"} {
+		if _, exists := config.Middleware[middlewareType]; !exists {
```

Comment:
Let's add an extended test (doesn't have to be in this PR, but close) that
starts a registry instance with a 1.2 and 1.3 config file in a config map
and actually starts a new registry instance and verifies the API comes up.

On Fri, Aug 26, 2016 at 12:58 PM, Jordan Liggitt <notifications@github.com>
wrote:

> In pkg/cmd/dockerregistry/dockerregistry.go
> <https://github.com/openshift/origin/pull/10673#discussion_r76451695>:
>
> > @@ -242,3 +248,31 @@ func panicHandler(handler http.Handler) http.Handler {
> >  		handler.ServeHTTP(w, r)
> >  	})
> >  }
> > +
> > +func setDefaultMiddleware(config *configuration.Configuration) (changed bool, err error) {
> > +	// Default to openshift middleware for relevant types
> > +	// This allows custom configs based on old default configs to continue to work
> > +	if config.Middleware == nil {
> > +		config.Middleware = map[string][]configuration.Middleware{}
> > +	}
> > +	for _, middlewareType := range []string{"registry", "repository", "storage"} {
> > +		if _, exists := config.Middleware[middlewareType]; !exists {
>
> if you're going to search and append anyway, you can drop this check...
> append will allocate it if needed
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/10673/files/ef3b4b7c799dfe9103f7fa63303c959d975f1ee7#r76451695>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p1Y_pnAJ17vm99uE05G4Dj-OBRhhks5qjxsxgaJpZM4JuLEY>
> .
>